cfy_manager/components/status_reporter/status_reporter.py

```diff
@@ -66,30 +63,21 @@ def configure(self):
 
     @staticmethod
     def _generate_node_id():
-        try:
-            with open(STATUS_REPORTER_CONFIGURATION_PATH) as f:
-                reporter_config = yaml.safe_load(f)
-        except yaml.YAMLError as e:
-            raise InitializationError('Failed loading status reporter\'s '
-                                      'configuration with the following: '
-                                      '{0}'.format(e))
-        reporter_config['node_id'] = str(uuid.uuid4())
-        updated_conf = yaml.safe_dump(reporter_config,
-                                      default_flow_style=False)
-        updated_conf_path = write_to_tempfile(updated_conf)
-        remove(STATUS_REPORTER_CONFIGURATION_PATH)
-        move(updated_conf_path, STATUS_REPORTER_CONFIGURATION_PATH)
-        chown('cfyreporter',
-              'cfyreporter',
-              STATUS_REPORTER_CONFIGURATION_PATH)
-        return reporter_config['node_id']
+        node_id = str(uuid.uuid4())
+        result = update_yaml_file(STATUS_REPORTER_CONFIGURATION_PATH,
+                                  'cfyreporter',
+                                  'cfyreporter',
+                                  {'node_id': node_id})
+        if not result:
+            raise InitializationError('Failed updating status reporter\'s '
+                                      'configuration.')
+        return node_id
 
     def remove(self):
         logger.notice('Removing status reporter {0}...'.format(
             self.reporter_type))
         systemd.remove(STATUS_REPORTER, service_file=False)
         yum_remove('cloudify_status_reporter')
         logger.info('Removing Status Reporter logs...')
-        remove('/var/log/status-reporter')
```

Comment:
You think we should keep the logs?

Reply:
This is a mistake, the logs are in a central place and handled like all other logs